### PR TITLE
[10.0][IMP] Refactor sepa_credit_transfert

### PR DESCRIPTION
**Pull request by p-tombez**
The goal is to allow overriding of line grouping (e.g. group by currency, partner, company, etc.)

### CODE REVIEW

**Comment by pedrobaeza** [comment]
You need to change module version number according https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#version-numbers and check Travis error.

**Review by leemannd — APPROVED** [review]

**Review by ghost — APPROVED** [review]
LGTM

**Review by ecino — APPROVED** [review]
LGTM. Can you review my PR #433 in return please?

File: account_banking_sepa_credit_transfer/models/account_payment_order.py [hunk 1]
```diff
@@ -11,6 +11,19 @@
 class AccountPaymentOrder(models.Model):
     _inherit = 'account.payment.order'
 
+    @api.model
+    def _get_line_key(self, line):
+        # The field line.date is the requested payment date
+        # taking into account the 'date_prefered' setting
+        # cf account_banking_payment_export/models/account_payment.py
+        # in the inherit of action_open()
+        return (
```
Comment (pedrobaeza) [inline account_banking_sepa_credit_transfer/models/account_payment_order.py:20]: Better to return a list, that it's dynamic

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Review by yvaucher — APPROVED** [review]
Appart from the comment from @pedrobaeza LGTM

File: account_banking_sepa_credit_transfer/models/account_payment_order.py (hunk 1 above)
Comment (p-tombez) [reply account_banking_sepa_credit_transfer/models/account_payment_order.py:20]: The returned value is used as a ```dict``` key so it needs to be "hashable"

Comment (pedrobaeza) [reply account_banking_sepa_credit_transfer/models/account_payment_order.py:20]: A list is also hashable

Comment (p-tombez) [reply account_banking_sepa_credit_transfer/models/account_payment_order.py:20]:
```
>>> {[1,2,3]: 'some value'}
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unhashable type: 'list'
```

Comment (jabibi) [reply account_banking_sepa_credit_transfer/models/account_payment_order.py:20]: It's okay like that if the main purpose is to use as a key. You can always go from tuple to list and from list to tuple if you want to modify the values

**Review by jabibi — APPROVED** [review]
Code review :+1:

**Comment by yvaucher** [comment]
@pedrobaeza can you update your review?

**Pull request merged**